selector/__init__.py

```diff
@@ -313,215 +324,207 @@ def visibility(self, site: Site, jobs=0, ephem_path=None, overwrite=False, sbtwo
             '''
             res = []
 
-            for id in tqdm(range(len(observations))):
-              
+            for obs in tqdm(observations):
                 res.append(self._calculate_visibility(site,
-                                                      observations[id].target.designation,
-                                                      observations[id].target.tag,
-                                                      observations[id].target.coordinates,
-                                                      observations[id].sky_conditions,
-                                                      observations[id].elevation,
-                                                      obs_windows[id],
+                                                      observations[obs.idx].target.designation,
```

Comment:
Better just use `obs.target.designation`

Reply:
Good point. With the loop change, for all of these we can just use `obs` instead of `observations[obs.idx]`.